exporters/variables-scss/src/generators/__tests__/fileGenerator.test.ts

```diff
@@ -41,43 +93,147 @@ describe('fileGenerator', () => {
+        { path: './js/global/', fileName: 'gradients.ts', content: emptyFile },
+        { path: './js/global/', fileName: 'typography.ts', content: emptyFile },
+        // Global index files
+        { path: './scss/global/', fileName: 'index.scss', content: indexFile },
+        { path: './js/global/', fileName: 'index.ts', content: indexJsFile },
+        // Root index files
+        { path: './scss/', fileName: '@global.scss', content: `@forward 'global';` },
+        { path: './js/', fileName: '@global.ts', content: `export * from './global';` },
+        // Themes files
+        { path: './scss/themes/theme-light/', fileName: '_colors.scss', content: emptyFile },
+        { path: './js/themes/theme-light/', fileName: 'colors.ts', content: emptyFile },
+        { path: './scss/themes/theme-light/', fileName: 'index.scss', content: indexColorFile },
+        { path: './js/themes/theme-light/', fileName: 'index.ts', content: indexJsColorFile },
+        { path: './scss/themes/theme-light-inverted/', fileName: '_colors.scss', content: emptyFile },
+        { path: './js/themes/theme-light-inverted/', fileName: 'colors.ts', content: emptyFile },
+        { path: './scss/themes/theme-light-inverted/', fileName: 'index.scss', content: indexColorFile },
+        { path: './js/themes/theme-light-inverted/', fileName: 'index.ts', content: indexJsColorFile },
+        // Themes root index files
+        { path: './scss/', fileName: '@themes.scss', content: mockedRootThemeFile },
+        { path: './js/', fileName: '@themes.ts', content: mockedRootThemeJsFile },
       ]);
     });
   });
 
   describe('generateFiles', () => {
     it('should generate files', () => {
-      const tokens = Array.from(exampleMockedTokens.values());
-      const files = generateFiles(tokens, mappedTokens, tokenGroups, nonThemedFilesData);
+      const tokens = Array.from(exampleDimensionAndStringTokens.values());
+      const files = generateFiles(tokens, mappedTokens, tokenGroups, nonThemedFilesData, false);
 
       expect(files).toStrictEqual([
         {
-          fileName: '_borders.scss',
+          fileName: 'borders',
           content: emptyFile,
         },
-        { fileName: '_other.scss', content: mockedExpectedResult },
-        { fileName: '_radii.scss', content: emptyFile },
-        { fileName: '_spacing.scss', content: emptyFile },
-        { fileName: '_shadows.scss', content: emptyFile },
-        { fileName: '_gradients.scss', content: emptyFile },
-        { fileName: '_typography.scss', content: emptyFile },
+        { fileName: 'other', content: mockedExpectedResult },
+        { fileName: 'radii', content: emptyFile },
+        { fileName: 'spacing', content: emptyFile },
+        { fileName: 'shadows', content: emptyFile },
+        { fileName: 'gradients', content: emptyFile },
+        { fileName: 'typography', content: emptyFile },
       ]);
     });
   });
+
+  describe('generateIndexFile', () => {
```

Comment:
The correct term for the "index" files is "barrel" file. I suggest using it instead of this name. Try to search "barrel files javascript".

Reply:
@literat does this also apply to scss? Because otherwise I'll have to name it something like generateIndexOrBarrelFile, since this generator handle both types

Reply:
The "barrel" file is a general term for all files that are used for re-exporting functions or libraries. So in our case, this is applicable to both JS design tokens and SCSS design tokens. The only difference is the technology itself - SCSS using `@forward` and JS using `export` - but still the way how we use the `index.scss|ts` is called the barrel file.